packages/identity/src/identity/delegation.ts

```diff
@@ -177,6 +194,38 @@ export class DelegationChain {
       targets?: Principal[];
     } = {},
   ): Promise<DelegationChain> {
+    /**
+     * Validations
+     */
+
+    // Validate expiration
+    if (expiration.getTime() < Date.now()) {
+      throw new DelegationError('Delegation expiration date cannot be in the past.');
+    }
+    if (options.previous) {
+      // Ensure we aren't extending beyond the maximum delegation chain length.
+      if (options.previous.delegations.length >= MAXIMUM_DELEGATION_CHAIN_LENGTH) {
+        throw new DelegationError(
+          `Delegation chain cannot exceed ${MAXIMUM_DELEGATION_CHAIN_LENGTH}`,
+        );
+      }
+
+      // Ensure that public keys are not repeated in the chain.
+      options.previous.delegations.reduce((previous, current) => {
```

Comment:
Given that `options.previous` is a DelegationChain, is it really necessary to check that `options.previous.delegations` has no repeated delegations?  

Asked another way, does `DelegationChain` have an invariant that `delegations` has no repeated public keys?

If so, it should be sufficient (and simpler) here to check that `to` is not found among `options.previous.delegations`.

Reply:
If TypeScript was the runtime language of the web, we could be confident that the other chain is a valid DelegationChain. However, if we want runtime validation, we have to do the checks ourself

Reply:
good thing MAXIMUM_DELEGATION_CHAIN_LENGTH is only 20.  Pretty sure this is O(n^2).